libs/core/algorithms/include/hpx/parallel/algorithms/adjacent_find.hpp

```diff
@@ -193,13 +193,9 @@ namespace hpx { namespace parallel { inline namespace v1 {
                 auto f1 = [pred_projected = HPX_MOVE(pred_projected), tok](
                               zip_iterator it, std::size_t part_size,
                               std::size_t base_idx) mutable {
-                    util::loop_idx_n<std::decay_t<ExPolicy>>(base_idx, it,
-                        part_size, tok,
-                        [&pred_projected, &tok](reference t, std::size_t i) {
-                            using hpx::get;
-                            if (pred_projected(get<0>(t), get<1>(t)))
-                                tok.cancel(i);
-                        });
+                    sequential_adjacent_find<std::decay_t<ExPolicy>>(base_idx,
+                        it, part_size, tok,
+                        std::forward<decltype(pred_projected)>(pred_projected));
```

Comment:
Forwarding `pred_projected` may create problems as the lambda `f1` may be called more than once.